tests/python/openassetio/hostAPI/test_manager.py

```diff
@@ -85,24 +78,19 @@ def some_refs():
 class Test_Manager_init:
 
     def test_interface_returns_the_constructor_supplied_object(
-            self, mock_manager_interface, host_session):
+            self, mock_manager_interface, mock_host_session):
 
         # pylint: disable=protected-access
-        a_manager = Manager(mock_manager_interface, host_session)
+        a_manager = Manager(mock_manager_interface, mock_host_session)
         assert a_manager._interface() is mock_manager_interface
 
     def test_when_constructed_with_ManagerInterface_as_None_then_raises_TypeError(
-            self, host_session):
+            self, mock_host_session):
         with pytest.raises(TypeError) as err:
-            Manager(None, host_session)
+            Manager(None, mock_host_session)
 
-        assert str(err.value) == (
-            '__init__(): incompatible constructor arguments. The following argument types '
-            'are supported:\n'
-            '    1. openassetio._openassetio.hostAPI.Manager(managerInterface: '
-            'openassetio._openassetio.managerAPI.ManagerInterface)\n'
-            '\n'
-            'Invoked with: None')
+        # Will contain a repr of the mock HostSession instance including address
+        assert str(err.value).startswith('__init__(): incompatible constructor arguments.')
```

Comment:
[consider] Might be worth also asserting the expected types are reported, i.e. `openassetio._openassetio.hostAPI.Manager` and possibly `openassetio._openassetio.managerAPI.ManagerInterface`. 

This then tests we get the order of `register*` functions correct in the bindings.  If the order is wrong then these won't be reported as the Python types (IIRC it'll use `typeinfo` and be all messy C++ types - see [pybind docs](https://pybind11.readthedocs.io/en/stable/advanced/misc.html#avoiding-c-types-in-docstrings)).

There is a handy keyword argument to `pytest.raises`, `match="<regex>"`, which could help.

Used in `test_context.py` for a similar reason.

Reply:
Updated [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/450/commits/1797f6a9e0c98c524fff42fd912813cd661446ed).

Reply:
Do you prefer the full error message rather than stripping the cruft via a regex using `match=`?

Reply:
How do you find [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/450/commits/4b6a5bdc3a6db4795567a424ddcb6e8fde083753) for readability?

Reply:
Sorry I missed the focus on type presentation. I think its also important to check that the message includes some illustration that the types are wrong. How about [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/450/commits/c4c16388bb839c5f486b672b70bb46010fae3ffe) (+fix [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/450/commits/34682b4fcc2b730971e14b568cdf426054abcb75)).